### PR TITLE
Site profiler: Language support

### LINKED ISSUE #82680: [Bug] Site profiler: Configure router to support languages
Support `/:lang/site-profiler` routes

### PR DESCRIPTION
**Pull request by bogiii**
Closes https://github.com/Automattic/wp-calypso/issues/82680

## Proposed Changes

* Supported language provided via route
* Refactored routing configuration

## Testing Instructions

* Go to `/site-profiler`
* Change the language from the Language chooser dropdown in the footer (for example Deutsch)
* Check if language has been change after the page re-render

- Go to `/de/site-profiler` or `/site-profiler/de`
- Check if language has been change

* Go to `/de/site-profiler/asdf.com` or `/site-profiler/de/asdf.com`
* Check if router handled route as before

- Go to `/de/site-profiler?domain=asdf.com` or `/site-profiler/de?domain=asdf.com`
- Check if router handled route as before

* Got to `/ar/site-profiler`
* Check if globe background is flipped

**Before:**
Screenshot 2023-10-10 at 12 45 49
**After:**
Screenshot 2023-10-10 at 12 43 11

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by zaerl — CHANGES_REQUESTED** [review]
The PR is indeed working. There is one small detail:

1. Go to http://calypso.localhost:3000/site-profiler/example.com
2. On the footer select a language, example `it`
3. You are redirected to `http://calypso.localhost:3000/site-profiler/it/site-profiler/example.com` with a validation error
4. It continues to append the string if you try again.

image

**Comment by bogiii** [comment]
Hey @zaerl,

The issue you reported has been fixed. Feel free to take another look.

Thanks!!!

File: client/site-profiler/controller.tsx [hunk 1]
```diff
@@ -18,20 +27,33 @@ export function redirectSiteProfilerResult( context: PageJS.Context, next: () =>
...
+		window.location.reload();
+	};
 
 	context.primary = (
 		<BrowserRouter>
 			<Main fullWidthLayout>
 				<SiteProfiler />
 			</Main>
-			{ isLoggedIn && <PureUniversalNavbarFooter isLoggedIn={ isLoggedIn } /> }
+			<PureUniversalNavbarFooter isLoggedIn={ isLoggedIn } onLanguageChange={ onLanguageChange } />
```
Comment (bogiii) [inline client/site-profiler/controller.tsx:56]: Now, we control the UnversalFooter component for both logged in/out contexts.

**Review by zaerl — APPROVED** [review]
LGTM.

**Pull request merged**